crates/jit-icache-coherence/Cargo.toml

```diff
@@ -20,4 +20,9 @@ features = [
 ]
 
 [target.'cfg(any(target_os = "linux", target_os = "macos"))'.dependencies.libc]
-version = "0.2.42"
\ No newline at end of file
+version = "0.2.42"
+
+[features]
+# Most modern cpu are smp,when This is enable some certain optimization can perform.
+# like riscv can use `fence.i` instead of linux syscall `sys_riscv_flush_icache`.
```

Comment:
Writing/grammar fix: let's reword this as

```
# Most modern CPUs are SMP (multicore). However, when only one core is present,
# some aspects of coherence are much cheaper. For example, RISC-V can use
# one instruction `fence.i` rather than a syscall that invokes all other cores.
# This feature enables such optimizations, but the resulting program will *only*
# be safe to run on one-core systems.
```

Reply:
I may have missed some discussion on this, but how come this is a crate feature? It seems like this would be better detected by the crate dynamically since many folks compiling code aren't certain what the target system looks like.

Reply:
There a comment about detecting the number of cpu's at runtime: https://github.com/bytecodealliance/wasmtime/pull/5078#discussion_r1000502204

I'd prefer if we could call the [vDSO implementation](https://github.com/torvalds/linux/blob/5bfc75d92efd494db37f5c4c173d3639d4772966/arch/riscv/kernel/vdso/flush_icache.S#L10), since that would also do the exact same thing for non SMP kernels, and avoid the syscalls.

Reply:
Could the result of `num_cpus::get` be stored in a `static`? That would incur syscalls on the first hit but none subsequently.

Reply:
@alexcrichton That is a good idea save `num_cpus::get`  in `static`.

Reply:
I don't think `num_cpus::get()` is safe. Linux supports hotplugging entire cpus and enabling/disabling cpu cores at runtime. In addition cgroups can be used to limit and unlimit cpu usage. `num_cpus` considers a cgroup limit to 1 or less cpu worth of time as there being only a single core even if it can migrate to a different cpu.

Reply:
@bjorn3  I have rollback to features implementation since not safe,safety come first.